async/retry_test.ts

```diff
@@ -31,11 +31,90 @@ Deno.test("[async] retry fails after max errors is passed", async function () {
   );
 });
 
-Deno.test("[async] retry throws if minTimeout is less than maxTimeout", async function () {
-  await assertRejects(() =>
-    retry(() => {}, {
+Deno.test(
+  "[async] retry throws if minTimeout is less than maxTimeout",
+  async function () {
+    await assertRejects(() =>
+      retry(() => {}, {
+        minTimeout: 1000,
+        maxTimeout: 100,
+      })
+    );
+  },
+);
+
+Deno.test(
+  "[async] retry does not throw if maxTimeout is -1",
+  async function () {
+    await retry(() => {}, {
       minTimeout: 1000,
-      maxTimeout: 100,
-    })
-  );
+      maxTimeout: -1,
+    });
+  },
+);
+
+// test util to ensure deterministic results during testing of backoff function by polyfilling Math.random
+function prngMulberry32(seed: number) {
+  return function () {
+    let t = (seed += 0x6d2b79f5);
+    t = Math.imul(t ^ (t >>> 15), t | 1);
+    t ^= t + Math.imul(t ^ (t >>> 7), t | 61);
+    return ((t ^ (t >>> 14)) >>> 0);
+  };
+}
+
+// random seed generated with crypto.getRandomValues(new Uint32Array(1))[0]
+const INITIAL_SEED = 3460544849;
+
+const firstTenExpectedTimings: {
+  results: readonly (readonly number[] & { length: 10 })[] & { length: 10 };
+  total: number;
+} = {
+  results: [
+    [69, 83, 56, 791, 131, 2140, 5480, 7706, 6036, 17908],
+    [54, 16, 23, 381, 145, 2717, 3195, 4374, 3149, 21390],
+    [32, 183, 334, 155, 391, 2954, 2890, 8202, 25202, 38387],
+    [54, 89, 26, 174, 741, 1245, 1021, 12191, 19834, 17559],
+    [74, 71, 113, 43, 496, 3196, 3843, 7860, 8943, 44312],
+    [20, 129, 52, 555, 857, 3072, 3955, 7078, 5640, 1339],
+    [75, 154, 59, 302, 998, 851, 5034, 8401, 23920, 41925],
+    [86, 26, 211, 491, 139, 2263, 4502, 10713, 15976, 32328],
+    [35, 10, 18, 449, 774, 698, 743, 8833, 24537, 7446],
+    [11, 122, 178, 132, 573, 1803, 5107, 4505, 11523, 17598],
+  ] as const,
+  total: 532_705,
+};
+
+Deno.test("[async] retry - backoff function timings", async (t) => {
+  const originalMathRandom = Math.random;
+
+  await t.step("_exponentialBackoffWithJitter", () => {
+    let nextSeed = INITIAL_SEED;
+
+    const expectedTimingData = firstTenExpectedTimings;
+
+    let total = 0;
+
+    for (const row of expectedTimingData.results) {
+      const randUint32 = prngMulberry32(nextSeed);
+      nextSeed = prngMulberry32(nextSeed)();
+      Math.random = () => randUint32() / 0x100000000;
+
+      const results: number[] = [];
+      const base = 100;
+      const cap = Infinity;
+
+      for (let i = 0; i < 10; ++i) {
+        const result = _exponentialBackoffWithJitter(cap, base, i, 2);
+        total += Math.round(result);
+        results.push(Math.round(result));
+      }
+
+      assertEquals(results as unknown, row);
+    }
+
+    assertEquals(total, expectedTimingData.total);
```

Comment:
Asserting total value feels redundant to me

Reply:
Removed